### DISCUSSION TITLE
Custom function to validate User SignUp process

### DISCUSSION THREAD

**Discussion by tobilab** [supabase/supabase#6385] [Questions] [answered]
Hello,
iam working on my first project with supabase and i need help to get the following functionality working:

At the Signup process to create new users i want to trigger custom condition-testing, e.g. concrete case: SignUp with email and password i want to ensure, that only users with a specific domain are able to signup.

Does anyone can pleaes help me to get the above mentioned functionality to work?
thanks in advance.

**Comment by GaryAustin1** [comment, marked as answer]
For just blocking on email domain, this general method should work.
https://github.com/supabase/supabase/discussions/5088

You could also use the new Edge Functions to handle anything you need to in the signup process.

**Reply by tobilab** [reply]
Thank you, i overread this - exactly my problem.

**Reply by dev-pandor** [reply]
Hey!. Just curious. How can I invoke an edge function right before a signup process? Should I create a trigger? Thanks

**Comment by tobilab** [comment]
> For just blocking on email domain, this general method should work.
> https://github.com/supabase/supabase/discussions/5088
> 
> You could also use the new Edge Functions to handle anything you need to in the signup process.

I searched for Tutorials, Docs and explanations on how to call a custom function at inserting content in a table but i dont understand that, do you have an example or further links for Help to get this realized? Because the function would need the request data to validate the typed in mail address and i dont have found any example for such a task.

**Reply by GaryAustin1** [reply]
I have this function from some testing I was doing awhile back... I think it works for this:
```
create function handle_check_user() returns trigger
    security definer
    SET search_path = public
    language plpgsql
as
$$
begin
if (split_part(new.email, '@', 2) = 'mydomain.com')
  then return new;
  else return null;
end if;
end;
$$;

create trigger on_auth_user_created
  before insert on auth.users
  for each row execute procedure public.handle_check_user();
```
"before insert" is what calls the function before versus after like in examples setting up a public profile table.

**Reply by tobilab** [reply]
it works great, thanks for your help!!!
